src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/NativeMemoryTests.cs

```diff
@@ -0,0 +1,159 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+
+namespace System.Runtime.InteropServices.Tests
+{
+    public unsafe class NativeMemoryTests
+    {
+        [Fact]
+        public void AlignedAllocTest()
+        {
+            void* ptr = NativeMemory.AlignedAlloc(1, (uint)sizeof(nuint));
+            Assert.True(ptr != null);
+            NativeMemory.AlignedFree(ptr);
+        }
+
+        [Fact]
+        public void AlignedAllocOOMTest()
+        {
+            Assert.Throws<OutOfMemoryException>(() => NativeMemory.AlignedAlloc(nuint.MaxValue - ((uint)sizeof(nuint) - 1), (uint)sizeof(nuint)));
```

Comment:
Should this also test `AlignedAlloc(nuint.MaxValue - ((uint)sizeof(nuint) - 2), (uint)sizeof(nuint))` ?

Reply:
Yes, I had actually added this test already and forgot to actually push the commit. Its up now under an explicit test covering overflow with a comment explaining why it exists:
https://github.com/dotnet/runtime/pull/54006/commits/3175ede4db1ededd318c781eb773d09122b0d36d#diff-219d972a6a5287dfe49b2e7863f5542f907f97a580e854cc9b9599da8b7d59d5R38-R61